### DISCUSSION TITLE
A custom style in the WYSIWIG editor

### DISCUSSION THREAD

**Discussion by johnwoltman** [GetPublii/Publii#1604] [Help]
I wanted to add a custom style in a span tag, like <span class="my-custom-badge">.  I found this in the docs: https://getpublii.com/docs/website-settings.html#editors - is this what I need? I don't really understand what that page is trying to explain, but it *seems* relevant.

**Comment by candidexmedia** [comment]
This Gist might be helpful: https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c

It documents three ways of adding custom blocks / elements / styles to the toolbar.

**Reply by apowell656** [reply]
@candidexmedia I looked at your gists. So can I assume I can create custom blocks using your methods? I am thinking of migrating several client sites from Hugo and Pelican to Publii, where I they can add custom sections/blocks on pages. I would appreciate any input or advice you have. 
For the devs, I have been using other tools for a while but constantly peeking at Publii.

**Reply by GetPublii** [reply]
There is no way to create custom blocks yet, but the good news is that we are working on it. :) 
The first custom blocks handled by the theme will be delivered to the Blocks Editor soon.

**Reply by candidexmedia** [reply]
@apowell656 The methods described in the gist are for the WYSIWYG Editor (which uses TinyMCE).

The [template plugin](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c#file-2-adding-a-template-plugin-for-reusable-chunks-md) is the most versatile option IMO for external clients. That said, it's more of a workaround that other Publii users figured out and experimented with using the TinyMCE docs, and it's not exactly documented on Publii's side.

The other two methods (Elements and Formats) have some documentation on Publii's site, but it isn't very clear.

Hopefully, custom blocks in the Blocks Editor will be an ever better option than the Template plugin in the WYSIWYG editor.

**Reply by candidexmedia** [reply]
Another thing to look into is custom helpers or the Smart Replace Plugin

See:
- https://github.com/GetPublii/Publii/discussions/1423#discussioncomment-8794636
-  this old forum post: https://forum.getpublii.com/topic/global-string-variable/

Essentially: instead of having to paste in the entire HTML for the bookmark and sift through it to replace the images and text, you could create some kind of custom shorthand, and tell Publii to find, say, `$foo` and replace it with the necessary code.

**Reply by apowell656** [reply]
@candidexmedia thanks for the response, I really appreciate it. I am going to have to make sure when I rework the designs I keep this information in consideration.

**Reply by candidexmedia** [reply]
Hi @GetPublii: when can we expect custom blocks for the Block Editor to be available?

**Reply by apowell656** [reply]
@GetPublii, I just converted one site to Tina CMS primarily because of block-editing and testing, but I have three more to go. Is it possible to do anything similar to that with what's here in Publii? 
I would really like to use Publii, but I also need to make sure small edits to landing pages can be handled by clients. Looking through the documentation, do you think or recommend partials could do something similar?

**Comment by johnwoltman** [comment]
Thank you! I will try that out
